lib/linguist/languages.yml

```diff
@@ -1495,23 +1495,35 @@ Forth:
   codemirror_mode: forth
   codemirror_mime_type: text/x-forth
   language_id: 114
-Fortran:
+Fortran Fixed Source:
+  group: Fortran
```

Comment:
Or we create an empty (no extension) Fortran entry and make both dialects children of that entry. Although since these names won't be displayed anyway, it probably would be easier to just rename one of the dialects to Fortran...

Reply:
> Or we create an empty (no extension) Fortran entry

I wasn't aware that was even possible. 😂

> it probably would be easier to just rename one of the dialects to Fortran...

Except that's probably going to cause a mess with repositories currently classified as Fortran, which won't have their language stats updated until a change is made. For historic or archived repositories, that probably isn't going to happen. Conversely, if we make the current Fortran an extension-less entry, no reclassification is necessary.

Reply:
I don't think a reclassification will be needed if we are also careful to keep the ID in place.

Reply:
That's what I'm saying. Let's keep the current entry as-is, sans any file extensions, and make the two dialects child-entries. 👍

Reply:
But even if the children are not visible to users, if we do that, all Fortran files will need to be updated to one of the children. Until they are, they will remain classified under the parent language (i.e., have ID 107 assigned) and will lose their syntax highlighting (unless we set a tm_scope for the parent language, but in that case, we might as well make the children with that same tm_scope the parent).

Reply:
> unless we set a `tm_scope` for the parent language

That's what I'm saying we should do.

> but in that case, we might as well make the children with that same `tm_scope` the parent

In the interest of fairness, it's probably better to have each dialect clearly named, rather than one "Fortran" and one "Fortran Fixed Form" (or "Fortran" + "Fortran Free Form", depending on which should be deemed the "real" Fortran…)

Reply:
As long as we don't break anything, I don't really care either way (+ users won't see these names anyway).

@lildude Any preference? One option involves a gradual reclassification of all ID-107 files, but I'm guessing that not an issue on GitHub's side...?

Reply:
Sorry about the delay.

Thinking and reading up on Fortran, I think the best approach to reduce change and confusion is to keep the current `Fortran` entry named as-is but only associate the "old" fixed-form extensions with this entry. This would effectively be the "original fixed form" Fortran.

Add a new "Fortran Free Source" or "Fortran Free Form" entry (the name isn't important as it won't be seen), with a new ID and no colour, with the new free-form extensions and grammar and make it a member of the "Fortran" group.

This way all current Fortran projects will remain identified as such and thanks to the caching, the users won't notice the difference either. All new projects and those actively updated will find either nothing will change (old fixed form) or they'll get improved syntax highlighting (new free form).